### PR TITLE
ESQL: Fixup docs for LOG and LOG10

### PR DESCRIPTION
**Pull request by nik9000**
This merges all of the hand written docs for `LOG` and `LOG10` into the annotations which updates the `META FUNCTIONS` - now it'll always be the same as the docs. This also deletes the hand maintained docs and let's the documentation generation process rebuild it.

### CODE REVIEW

File: docs/reference/esql/functions/description/log.asciidoc [hunk 1]
```diff
@@ -2,4 +2,4 @@
 
 *Description*
 
-Returns the logarithm of a number to a base.
+Returns the logarithm of a value to a base. The input can be any numeric value, the return value is always a double.  Logs of zero, negative numbers, infinities and base of one return `null` as well as a warning.
```
Comment (not-napoleon) [inline docs/reference/esql/functions/description/log.asciidoc:5]: Nit: I don't think there's any way (or at least, there shouldn't be any non-bug way) to generate an infinite in ES|QL.  I believe we always turn them into nulls instead.

File: x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/math/Log10.java [hunk 2]
```diff
@@ -28,8 +29,21 @@
 import static org.elasticsearch.xpack.ql.expression.TypeResolutions.isNumeric;
 
 public class Log10 extends UnaryScalarFunction {
-    @FunctionInfo(returnType = "double", description = "Returns the log base 10.")
-    public Log10(Source source, @Param(name = "number", type = { "double", "integer", "long", "unsigned_long" }) Expression n) {
+    @FunctionInfo(
+        returnType = "double",
+        description = "Returns the logarithm to base 10. The input can be any numeric value, the return value is always a double.\n"
```
Comment (leemthompo) [inline x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/math/Log10.java:34]:
```suggestion
        description = "Returns the logarithm to base 10 of a value. The input can be any numeric value, the return value is always a double.\n"
```

nit for consistency

**Review by not-napoleon — APPROVED** [review]
LGTM

**Review by leemthompo — COMMENTED** [review]
Looks good on docs side, with one `nit`

**Pull request merged**